### PR TITLE
implement ProxySourceService

### PR DESCRIPTION
**Pull request by williambrownqqq**
implement ProxySourceService

ProxySourceService utilize FileParser to get data from files.
and as a result we retain all proxyConfigs in Queue<ProxyConfigHolder>, where ProxyConfigHolder represents entity with ProxyCredentials and ProxyNetworkConfig

### CODE REVIEW

File: src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public ProxyConfigHolder getProxy(){
+        ProxyConfigHolder proxyConfigHolder = proxyConfigHolders.peek();
+
+        synchronized (this){
+            if(proxyConfigHolder == null){
+                proxyConfigHolders.addAll(getAllProxyConfigs());
+            }
+        }
+        return proxyConfigHolders.poll();
+    }
```
Comment (MazurenkoNick) [inline src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java:43]: to delete

File: src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public class ProxySourceServiceFile implements ProxySourceService {
+    private static final String PROXY_CONFIGS_PATH = "json/ProxyConfigs.json";
+    private final FileParser fileParser;
+    private final BlockingQueue<ProxyConfigHolder> proxyConfigHolders;
+    @Autowired
+    public ProxySourceServiceFile(FileParser fileParser) {
+        this.fileParser = fileParser;
+        this.proxyConfigHolders = new LinkedBlockingDeque<>();
+    }
+
```
Comment (MazurenkoNick) [inline src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java:25]:
```suggestion
@Override
```

File: src/main/java/com/gfl/client/service/proxy/ProxySourceService.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.gfl.client.service.proxy;
+
+import com.gfl.client.model.dto.ProxyConfigHolder;
+
+public interface ProxySourceService {
+        ProxyConfigHolder getProxy();
```
Comment (MazurenkoNick) [inline src/main/java/com/gfl/client/service/proxy/ProxySourceService.java:6]:
```suggestion
        List<ProxyConfigHolder> getAllProxyConfigs();
```

File: src/main/java/com/gfl/client/model/dto/ProxyConfigHolder.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package com.gfl.client.model.dto;
+
+import java.util.Objects;
+
```
Comment (MazurenkoNick) [inline src/main/java/com/gfl/client/model/dto/ProxyConfigHolder.java:4]: Use lombok instead of explicit getters, setters, etc.

**Review by MazurenkoNick — CHANGES_REQUESTED** [review]

**Review by MazurenkoNick — APPROVED** [review]

File: src/main/java/com/gfl/client/model/dto/ProxyConfigHolder.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.gfl.client.model.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
```
Comment (Lokankara) [inline src/main/java/com/gfl/client/model/dto/ProxyConfigHolder.java:8]: `@Data` already contains ` @RequiredArgsConstructor`
You can remove `@NoArgsConstructor, @AllArgsConstructor`

File: src/main/java/com/gfl/client/model/dto/ProxyCredentials.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package com.gfl.client.model.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Objects;
```
Comment (Lokankara) [inline src/main/java/com/gfl/client/model/dto/ProxyCredentials.java:7]: Unused import statement

File: src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+
+@Service
+public class ProxySourceServiceFile implements ProxySourceService {
+    private static final String PROXY_CONFIGS_PATH = "json/ProxyConfigs.json";
+    private final FileParser fileParser;
+    private final BlockingQueue<ProxyConfigHolder> proxyConfigHolders;
```
Comment (Lokankara) [inline src/main/java/com/gfl/client/service/proxy/ProxySourceServiceFile.java:19]: Private field 'proxyConfigHolders' is assigned but never accessed

**Review by Lokankara — APPROVED** [review]

**Pull request merged**